src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs

```diff
@@ -158,6 +160,25 @@ internal sealed class EvaluationContext : EvaluationContextBase
             Guid moduleVersionId,
```

Comment:
Consider:  Can we delete this overload and update the callers to pass "previous == null" to the overload above instead?  It's not a big deal, but it seems like a bit of a gotcha that we have to do the ```NormalizeILOffset``` check in each factory method.